text/0097-log-data-model.md

```diff
@@ -0,0 +1,972 @@
+  * [Field: `Timestamp`](#field-timestamp)
+  * [Trace Context Fields](#trace-context-fields)
+    * [Field: `TraceId`](#field-traceid)
+    * [Field: `SpanId`](#field-spanid)
+    * [Field: `TraceFlags`](#field-traceflags)
+  * [Severity Fields](#severity-fields)
+    * [Field: `SeverityText`](#field-severitytext)
+    * [Field: `SeverityNumber`](#field-severitynumber)
+    * [Displaying Severity](#displaying-severity)
+    * [Comparing Severity](#comparing-severity)
+  * [Field: `ShortName`](#field-shortname)
+  * [Field: `Body`](#field-body)
+  * [Field: `Resource`](#field-resource)
+  * [Field: `Attributes`](#field-attributes)
+* [Example Log Records](#example-log-records)
+* [Open Questions](#open-questions)
+* [Alternate Design](#alternate-design)
+* [Prior Art](#prior-art)
+  * [RFC5424 Syslog](#rfc5424-syslog)
+  * [Fluentd Forward Protocol Model](#fluentd-forward-protocol-model)
+* [Appendix A. Example Mappings.](#appendix-a-example-mappings)
+  * [RFC5424 Syslog](#rfc5424-syslog-1)
+  * [Windows Event Log](#windows-event-log)
+  * [SignalFx Events](#signalfx-events)
+  * [Splunk HEC](#splunk-hec)
+  * [Log4j](#log4j)
+  * [Zap](#zap)
+  * [Apache](#apache)
+  * [CloudTrail Log Event](#cloudtrail-log-event)
+* [Appendix B: `SeverityNumber` example mappings.](#appendix-b-severitynumber-example-mappings)
+* [References](#references)
+
+## Motivation
+
+This is a proposal of a data model and semantic conventions that allow to
+represent logs from various sources: application log files, machine generated
+events, system logs, etc. Existing log formats can be unambiguously mapped to
+this data model. Reverse mapping from this data model is also possible to the
+extent that the target log format has equivalent capabilities.
+
+The purpose of the data model is to have a common understanding of what a log
+record is, what data needs to be recorded, transferred, stored and interpreted
+by a logging system.
+
+## Design Notes
+
+### Requirements
+
+The Data Model was designed to satisfy the following requirements:
+
+- It should be possible to unambiguously map existing log formats to this Data
+  Model. Translating log data from an arbitrary log format to this Data Model
+  and back should ideally result in identical data.
+
+- Mappings of other log formats to this Data Model should be semantically
+  meaningful. The Data Model must preserve the semantics of particular elements
+  of existing log formats.
+
+- Translating log data from an arbitrary log format A to this Data Model and
+  then translating from the Data Model to another log format B ideally must
+  result in a meaningful translation of log data that is no worse than a
+  reasonable direct translation from log format A to log format B.
+
+- It should be possible to efficiently represent the Data Model in concrete
+  implementations that require the data to be stored or transmitted. We
+  primarily care about 2 aspects of efficiency: CPU usage for
```

Comment:
should the "addition of attributes" be an aspect of the efficiency?

Reply:
Sorry, I am not sure I understand what "addition of attributes" is.

Reply:
I meant enriching with more attributes on different stages of processing.

Reply:
Yes, definitely. That should be considered when working on the representation of this data model, e.g. when adding logs support to OTLP. However, I do not think it enrichment considerations have a special effect on the data model design.

Reply:
Resolving. Feel free to reopen if needed.